File: backend/src/main/java/com/godeltech/mastery/backend/repository/ClientRepository.java
package com.godeltech.mastery.backend.repository;

import com.godeltech.mastery.backend.domain.entity.Client;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;

public interface ClientRepository extends JpaRepository<Client, Long> {

  Boolean existsByEmailIgnoreCase(String email);

  Optional<Client> getByEmailIgnoreCase(String email);
}

File: backend/src/main/kotlin/com/godeltech/mastery/backend/repository/ClientRepository.kt
package com.godeltech.mastery.backend.repository

import com.godeltech.mastery.backend.domain.entity.Client
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface ClientRepository : JpaRepository<Client, Long> {

    fun existsByEmailIgnoreCase(email: String): Boolean

    fun getByEmailIgnoreCase(email: String): Optional<Client>
}